packages/react-label/src/components/Label/useLabel.tsx

```diff
@@ -18,40 +17,18 @@ const mergeProps = makeMergeProps<LabelState>({ deepMerge: labelShorthandProps }
  *
  * @param props - props from this instance of Label
  * @param ref - reference to root HTMLElement of Label
- * @param defaultProps - (optional) default prop values provided by the implementing type
  */
-export const useLabel = (props: LabelProps, ref: React.Ref<HTMLElement>, defaultProps?: LabelProps): LabelState => {
-  const state = mergeProps(
-    {
-      ref,
-      as: 'label',
-      size: 'medium',
-      required: {
-        as: 'span',
-      },
-    },
-    defaultProps && resolveLabelShorthandProps(defaultProps),
-    resolveLabelShorthandProps(props),
-  );
-
-  return state;
-};
-
-/**
- * Label will first need to check if required is a boolean or shorthandprops,
- * this allows for the required prop to handle both the default asterisk for required
- * or a custom required text.
- */
-const resolveLabelShorthandProps = (props: LabelProps) => {
-  let propsNormalized;
-  if (props.required === true) {
-    propsNormalized = { ...props, required: { children: '*' } };
-  } else if (props.required === false) {
-    propsNormalized = { ...props, required: undefined };
-  } else {
-    // TypeScript needs a nudge to figure out that props.required won't be a boolean here
-    propsNormalized = props as LabelProps & { required?: Exclude<LabelProps['required'], boolean> };
-  }
-
-  return resolveShorthandProps(propsNormalized, labelShorthandProps);
+export const useLabel = (props: LabelProps, ref: React.Ref<HTMLElement>): LabelState => {
+  const { disabled = false, required = false, strong = false, size = 'medium' } = props;
+  return {
+    disabled,
+    required: resolveShorthand(required === false ? null : required, {
+      required: !!required,
+      defaultProps: { children: '*' },
+    }),
```

Comment:
nit: Move this to the end of this return statement so that all slots are processed next to each other.